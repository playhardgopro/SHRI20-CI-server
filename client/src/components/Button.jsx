import React from 'react'
import PropTypes from 'prop-types'
import { withNaming } from '@bem-react/classname'
// import './scss/Button.scss'
const cn = withNaming({ n: '', e: '__', m: '_' })

const Button = ({ children, onClick, className, disabled, active }) => {
  const classes = cn('button')(className)

  return (
    <button type="button" className={classes} disable={disabled} onClick={onClick}>
      {children}
    </button>
    // <button type="button" className="button button_size_s button_distribute_center button_view_control">
    // <div className="icon icon_size_s">
    //   <svg width="12" height="12" viewbox="0 0 12 12" xmlns="http://www.w3.org/2000/svg">
    //     <path d="M8.77548 3.03097C8.01915 2.32277 7.03761 1.93456 5.99632 1.93548C4.12234 1.93713 2.50452 3.22205 2.05783 5.00444C2.02531 5.13419 1.90972 5.22581 1.77595 5.22581H0.389589C0.208185 5.22581 0.070379 5.06112 0.103935 4.88284C0.627411 2.103 3.06806 0 6 0C7.60761 0 9.06752 0.632322 10.1447 1.66173L11.0088 0.797661C11.3746 0.431879 12 0.690944 12 1.20825V4.45161C12 4.7723 11.74 5.03226 11.4194 5.03226H8.17599C7.65869 5.03226 7.39962 4.40683 7.7654 4.04102L8.77548 3.03097ZM0.580645 6.96774H3.82401C4.34131 6.96774 4.60038 7.59317 4.2346 7.95898L3.22452 8.96906C3.98085 9.67727 4.96246 10.0655 6.00377 10.0645C7.87679 10.0628 9.49527 8.7788 9.94217 6.99561C9.97468 6.86586 10.0903 6.77424 10.224 6.77424H11.6104C11.7918 6.77424 11.9296 6.93893 11.8961 7.11721C11.3726 9.897 8.93194 12 6 12C4.39239 12 2.93248 11.3677 1.85528 10.3383L0.99121 11.2023C0.625427 11.5681 0 11.3091 0 10.7917V7.54839C0 7.2277 0.25996 6.96774 0.580645 6.96774Z" />
    //   </svg>
    // </div>
    // <div className="button__text decorator hide_mobile">Rebuild</div>
    // </button>
  )
}

Button.propTypes = {
  children: PropTypes.node,
  onClick: PropTypes.func,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  active: PropTypes.bool,
}
Button.defaultProps = {
  children: 'Default button',
  onClick: () => {},
  className: '',
  disabled: false,
  active: false,
}

export default Button
