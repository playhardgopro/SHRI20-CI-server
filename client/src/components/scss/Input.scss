@import './var.scss';
.input {
  border: 2px solid var(--color-bg-border);
  border-radius: 4px;
  &:focus {
    border: 2px solid var(--color-bg-border-focused);
  }
}
.input:focus ~ .input__controls-append .button::after {
  border-color: var(--color-bg-border-focused);
}

.input ~ .input__controls-append .icon {
  color: var(--blue-grey-250);
}

.input ~ .input__controls-append .button:hover .icon {
  color: var(--color-bg-border-focused);
}

.input_size_m {
  height: var(--text-line-height-36);
  padding-right: var(--space-13);
  padding-left: var(--space-13);
}

.input_width_full {
  flex: 1 1 auto;
}
.input_width_52 {
  width: 52px;
  max-width: 100px;
}

.input__group {
  display: flex;
  align-items: center;
}

.input__group_vertical {
  align-items: stretch;
  flex-direction: column;
}

@media screen and (max-width: $screen-m) {
  .input__group_vertical.input__group_m-horizontal .input__label {
    margin-bottom: 0;
  }
}
.input__group_vertical .input__label {
  margin-right: 0;
  margin-bottom: var(--space-6);
}

.input__label {
  margin-right: var(--space-8);
}

.input__label_required::after {
  color: var(--red-600);
  content: ' *';
}

.input__controls {
  display: flex;
}

.input__controls-append {
  display: flex;
  align-items: center;
  margin-left: -2px;
}
.input__controls-append .text {
  margin-left: calc(var(--space-8) + 2px);
}

.input__controls-append .button {
  height: 100%;
  margin-left: -24px;
  background-color: var(--color-bg-main);
}
.input__controls-append .button:after {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  content: '';
  border: 2px solid var(--color-bg-border);
  border-left: 0;
  border-top-left-radius: 0;
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
  border-bottom-left-radius: 0;
}
.input:focus .input__controls-append .button::after {
  border: 2px solid var(--color-bg-border-focused);
}

.input__group > .input:not(:last-child) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}
.input:not(:placeholder-shown) + .input__controls-append .button {
  display: none;
}
