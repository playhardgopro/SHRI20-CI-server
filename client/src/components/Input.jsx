import React from 'react'
import PropTypes from 'prop-types'
import { withNaming } from '@bem-react/classname'
import { Text, Icon, Button } from '.'
import './scss/Input.scss'

const cn = withNaming({ n: '', e: '__', m: '_' })

const cnInput = cn('input')

const textStyle = { size: '13-18', type: 'h2' }

const LilInput = ({ id, placeholder, isShort }) => {
  const inputClass = { size: 'm', width: isShort ? 52 : 'full' }
  return (
    <input className={cnInput(inputClass, ['text', { size: '13-15' }])} id={id} type="text" placeholder={placeholder} />
  )
}

const ControlsAppend = ({ text }) => {
  return text ? (
    <Text className={{ size: '13-18' }}>minutes</Text>
  ) : (
    <button type="button" className="button button_size_m button_distribute_center button_view_control">
      <Icon name="close" className={{ size: 'm' }} />
    </button>
  )
}

const Input = ({ children, className, options }) => {
  const { placeholder, label, isRequired, id, vertical, text } = options
  return (
    <div className={cnInput('group', { vertical })}>
      <label className={cnInput('label', { required: isRequired })} htmlFor="repository">
        <Text className={textStyle}>{label}</Text>
      </label>
      <div className={cnInput('controls')}>
        <LilInput id={id} placeholder={placeholder} isShort={text} />
        <div className={cnInput('controls-append')}>{ControlsAppend({ text })}</div>
      </div>
    </div>
  )
}

Input.propTypes = {
  children: PropTypes.node,
  className: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.bool])),
}
Input.defaultProps = {
  children: '',
  className: {},
  options: { label: '', placeholder: '' },
}

export default Input
