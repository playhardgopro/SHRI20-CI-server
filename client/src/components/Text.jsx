import React from 'react'
import PropTypes from 'prop-types'
import { withNaming } from '@bem-react/classname'
import './scss/Text.scss'

const cn = withNaming({ n: '', e: '__', m: '_' })

const Text = ({ children, className }) => {
  const cnText = cn('text')

  return <span className={cnText(className)}>{children}</span>
}

Text.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  className: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.bool])),
}
Text.defaultProps = {
  children: '',
  className: {},
}

export default Text
