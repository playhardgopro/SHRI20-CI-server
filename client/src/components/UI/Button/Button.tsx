import * as React from 'react'
import PropTypes from 'prop-types'
import { withNaming } from '@bem-react/classname'
import { Icon } from '../../index'
import './Button.scss'

const cn = withNaming({ n: '', e: '__', m: '_' })

const Button = ({ children, onClick, className, disabled, icon, hideText, empty }) => {
  const cnButton = cn('button')
  const cnHide = [cnButton('text'), 'decorator', 'hide-mobile'].join(' ')

  return (
    <button type="button" className={cnButton(className)} disabled={disabled} onClick={onClick}>
      {icon && <Icon name={icon.name} className={{ size: icon.size }} />}
      {!empty && !hideText && <div className={cnButton('text')}>{children}</div>}
      {!empty && hideText && <div className={cnHide}>{children}</div>}
    </button>
  )
}

Button.propTypes = {
  children: PropTypes.node,
  onClick: PropTypes.func,
  className: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.bool])),
  disabled: PropTypes.bool,
  empty: PropTypes.bool,
  hideText: PropTypes.bool,
  icon: PropTypes.oneOfType([
    PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.bool])),
    PropTypes.bool,
  ]),
  // active: PropTypes.bool,
}
Button.defaultProps = {
  children: null,
  onClick: () => {},
  className: {},
  disabled: false,
  empty: false,
  hideText: false,
  icon: false,
  // active: false,
}

export default Button
