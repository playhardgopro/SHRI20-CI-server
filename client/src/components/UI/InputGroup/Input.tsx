import * as React from 'react'
import { withNaming } from '@bem-react/classname'
import { Text, Button } from '../../index'
import './InputGroup.scss'

const cn = withNaming({ n: '', e: '__', m: '_' })
const cnInput = cn('input')
const cnText = cn('text')

type name = 'repoName' | 'buildCommand' | 'mainBranch' | 'period'
interface InputProps {
  clearable?: boolean
  id: name
  name: name
  width?: 'full' | 52
  size?: 's' | 'm'
  placeholder: string
  onClear(name: name): void
  text?: string
  status: 'invalid' | 'valid' | undefined
  className?: {}
  inputRef: (ref: HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement | null) => void
}

export const Input: React.FC<InputProps> = ({
  id,
  name,
  clearable,
  onClear,
  inputRef,
  placeholder,
  status,
  className,
  text,
  width,
  size,
}) => {
  const handleClear = () => {
    onClear(name)
  }
  return (
    <div className={cnInput('controls')}>
      <input
        id={id}
        name={name}
        ref={inputRef}
        placeholder={placeholder}
        className={cnInput({ ...className, status, width, size }, [cnText({ size: '13-15' })])}
      />
      <div className={cnInput('controls-append')}>
        {clearable && (
          <Button
            className={{ size: 'm', distribute: 'center', view: 'control' }}
            icon={{ name: 'close', size: 'm' }}
            onClick={handleClear}
            empty
          />
        )}
        {text && <Text className={{ size: '13-18' }}>{text}</Text>}
      </div>
    </div>
  )
}

export default Input
