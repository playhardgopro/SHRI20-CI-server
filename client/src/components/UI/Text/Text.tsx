import * as React from 'react'
import { withNaming } from '@bem-react/classname'
import './Text.scss'

const cn = withNaming({ n: '', e: '__', m: '_' })
interface TextProps {
  className: {}
}

const Text: React.FC<TextProps> = ({ children, className }) => {
  const cnText = cn('text')

  return <span className={cnText(className)}>{children}</span>
}

export default Text
