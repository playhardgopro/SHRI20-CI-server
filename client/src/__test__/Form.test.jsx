import React, { useRef, useState } from 'react'
import renderer from 'react-test-renderer'
import { withNaming } from '@bem-react/classname'
import Input from '../components/UI/InputGroup/Input'
import Button from '../components/UI/Button/Button'
import Text from '../components/UI/Text/Text'

// jest.mock('react-text-mask', () => (props) => <input type="text" {...{ ...props }} />)

const Stub = () => {
  const cn = withNaming({ n: '', e: '__', m: '_' })
  const cnForm = cn('form')
  const cnInput = cn('input')
  const textStyle = { size: '13-18', type: 'h2' }

  const errors = {
    repoName: { message: 'Field is required' },
    buildCommand: { message: 'Field is required' },
    mainBranch: { message: 'Field is required' },
    period: { message: 'Field is required' },
  }
  // const formRef = useRef(null)
  return (
    <form className={cnForm()}>
      <div className={cnForm('title')}>
        <div className="form__header text text_type_h2 text_size_15-20">Settings</div>
        <div className="form__subheader text text_type_h3 text_size_13-18 text_view_ghost">
          Configure repository connection and synchronization settings.
        </div>
      </div>
      <div className={cnForm('items')}>
        <div className={cnForm('item', { 'indent-t': 'xl' })}>
          <div className={cnInput('group', { vertical: true })}>
            <label className={cnInput('label', { required: true })} htmlFor="repository">
              <Text className={textStyle}>Github repository</Text>
            </label>
            <Input
              id="repoName"
              placeholder="username/reponame"
              status={errors.repoName && 'invalid'}
              name="repoName"
              clearable
            />
            {errors.repoName && <Text className={{ ...textStyle, view: 'error' }}>{errors.repoName.message}</Text>}
          </div>
        </div>
        <div className={cnForm('item', { 'indent-t': 'xl' })}>
          <div className={cnInput('group', { vertical: true })}>
            <label className={cnInput('label', { required: true })} htmlFor="repository">
              <Text className={textStyle}>Build command</Text>
            </label>
            <Input
              id="buildCommand"
              placeholder="npm run build"
              status={errors.buildCommand && 'invalid'}
              name="buildCommand"
              clearable
            />
            {errors.buildCommand && (
              <Text className={{ ...textStyle, view: 'error' }}>{errors.buildCommand.message}</Text>
            )}
          </div>
        </div>
        <div className={cnForm('item', { 'indent-t': 'xl' })}>
          <div className={cnInput('group', { vertical: true })}>
            <label className={cnInput('label', { required: true })} htmlFor="repository">
              <Text className={textStyle}>Main branch</Text>
            </label>
            <Input
              id="mainBranch"
              placeholder="master"
              status={errors.mainBranch && 'invalid'}
              name="mainBranch"
              clearable
            />
            {errors.mainBranch && <Text className={{ ...textStyle, view: 'error' }}>{errors.mainBranch.message}</Text>}
          </div>
        </div>
        <div className={cnForm('item', { 'indent-t': 'xl' })}>
          <div className={cnInput('group', { vertical: false })}>
            <label className={cnInput('label', { required: false })} htmlFor="period">
              <Text className={textStyle}>Synchronize every</Text>
            </label>
            <Input
              id="period"
              placeholder="10"
              status={errors.period && 'invalid'}
              name="period"
              width="52"
              text="minutes"
            />
          </div>
          {errors.period && <Text className={{ ...textStyle, view: 'error' }}>{errors.period.message}</Text>}
          <Text className={{ ...textStyle, view: 'error' }}>Field is required</Text>
        </div>
      </div>
      <div className={cnForm('controls')}>
        <Button className={{ size: 'm', view: 'action' }} disabled={false}>
          Save
        </Button>
        <Button className={{ size: 'm', view: 'control' }} disabled={false}>
          Cancel
        </Button>
      </div>
    </form>
  )
}
describe('Form', () => {
  it('renders correctly', () => {
    const form = renderer.create(<Stub />).toJSON()

    expect(form).toMatchSnapshot()
  })
})
