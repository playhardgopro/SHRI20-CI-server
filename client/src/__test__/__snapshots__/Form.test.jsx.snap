// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Form correctly 1`] = `
<form
  className="form"
>
  <div
    className="form__title"
  >
    <div
      className="form__header text text_type_h2 text_size_15-20"
    >
      Settings
    </div>
    <div
      className="form__subheader text text_type_h3 text_size_13-18 text_view_ghost"
    >
      Configure repository connection and synchronization settings.
    </div>
  </div>
  <div
    className="form__items"
  >
    <div
      className="form__item form__item_indent-t_xl"
    >
      <div
        className="input__group input__group_vertical"
      >
        <label
          className="input__label input__label_required"
          htmlFor="repository"
        >
          <span
            className="text text_size_13-18 text_type_h2"
          >
            Github repository
          </span>
        </label>
        <div
          className="input__controls"
        >
          <input
            className="input input_status_invalid input_width_full input_size_m text text_size_13-15"
            id="repoName"
            name="repoName"
            placeholder="username/reponame"
          />
          <div
            className="input__controls-append"
          >
            <button
              className="button button_size_m button_distribute_center button_view_control"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span
                className="icon icon_size_m"
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 16C3.6 16 0 12.4 0 8C0 3.6 3.6 0 8 0C12.4 0 16 3.6 16 8C16 12.4 12.4 16 8 16ZM12 5.12L10.88 4L8 6.88L5.12 4L4 5.12L6.88 8L4 10.88L5.12 12L8 9.12L10.88 12L12 10.88L9.12 8L12 5.12Z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <span
          className="text text_size_13-18 text_type_h2 text_view_error"
        >
          Field is required
        </span>
      </div>
    </div>
    <div
      className="form__item form__item_indent-t_xl"
    >
      <div
        className="input__group input__group_vertical"
      >
        <label
          className="input__label input__label_required"
          htmlFor="repository"
        >
          <span
            className="text text_size_13-18 text_type_h2"
          >
            Build command
          </span>
        </label>
        <div
          className="input__controls"
        >
          <input
            className="input input_status_invalid input_width_full input_size_m text text_size_13-15"
            id="buildCommand"
            name="buildCommand"
            placeholder="npm run build"
          />
          <div
            className="input__controls-append"
          >
            <button
              className="button button_size_m button_distribute_center button_view_control"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span
                className="icon icon_size_m"
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 16C3.6 16 0 12.4 0 8C0 3.6 3.6 0 8 0C12.4 0 16 3.6 16 8C16 12.4 12.4 16 8 16ZM12 5.12L10.88 4L8 6.88L5.12 4L4 5.12L6.88 8L4 10.88L5.12 12L8 9.12L10.88 12L12 10.88L9.12 8L12 5.12Z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <span
          className="text text_size_13-18 text_type_h2 text_view_error"
        >
          Field is required
        </span>
      </div>
    </div>
    <div
      className="form__item form__item_indent-t_xl"
    >
      <div
        className="input__group input__group_vertical"
      >
        <label
          className="input__label input__label_required"
          htmlFor="repository"
        >
          <span
            className="text text_size_13-18 text_type_h2"
          >
            Main branch
          </span>
        </label>
        <div
          className="input__controls"
        >
          <input
            className="input input_status_invalid input_width_full input_size_m text text_size_13-15"
            id="mainBranch"
            name="mainBranch"
            placeholder="master"
          />
          <div
            className="input__controls-append"
          >
            <button
              className="button button_size_m button_distribute_center button_view_control"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span
                className="icon icon_size_m"
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 16C3.6 16 0 12.4 0 8C0 3.6 3.6 0 8 0C12.4 0 16 3.6 16 8C16 12.4 12.4 16 8 16ZM12 5.12L10.88 4L8 6.88L5.12 4L4 5.12L6.88 8L4 10.88L5.12 12L8 9.12L10.88 12L12 10.88L9.12 8L12 5.12Z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <span
          className="text text_size_13-18 text_type_h2 text_view_error"
        >
          Field is required
        </span>
      </div>
    </div>
    <div
      className="form__item form__item_indent-t_xl"
    >
      <div
        className="input__group"
      >
        <label
          className="input__label"
          htmlFor="period"
        >
          <span
            className="text text_size_13-18 text_type_h2"
          >
            Synchronize every
          </span>
        </label>
        <div
          className="input__controls"
        >
          <input
            className="input input_status_invalid input_width_52 input_size_m text text_size_13-15"
            id="period"
            name="period"
            placeholder="10"
          />
          <div
            className="input__controls-append"
          >
            <span
              className="text text_size_13-18"
            >
              minutes
            </span>
          </div>
        </div>
      </div>
      <span
        className="text text_size_13-18 text_type_h2 text_view_error"
      >
        Field is required
      </span>
      <span
        className="text text_size_13-18 text_type_h2 text_view_error"
      >
        Field is required
      </span>
    </div>
  </div>
  <div
    className="form__controls"
  >
    <button
      className="button button_size_m button_view_action"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <div
        className="button__text"
      >
        Save
      </div>
    </button>
    <button
      className="button button_size_m button_view_control"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <div
        className="button__text"
      >
        Cancel
      </div>
    </button>
  </div>
</form>
`;
